library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoPollableOpeartorTest.java

```diff
@@ -0,0 +1,251 @@
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultPartition;
+import com.datatorrent.api.Partitioner;
+import com.datatorrent.lib.helper.OperatorContextTestHelper;
+import com.datatorrent.lib.helper.TestPortContext;
+import com.datatorrent.lib.testbench.CollectorTestSink;
+import com.datatorrent.lib.util.FieldInfo;
+import com.datatorrent.lib.util.KeyValPair;
+import com.datatorrent.lib.util.TestUtils;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JdbcPojoPollableOpeartorTest extends JdbcOperatorTest
+{
+  public String dir = null;
+  @Mock
+  private ScheduledExecutorService mockschedular;
+  @Mock
+  private ScheduledFuture futureTaskMock;
+  @Mock
+  private WindowDataManager windowDataManagerMock;
+
+  @Before
+  public void beforeTest()
+  {
+    dir = "target/" + APP_ID + "/";
+
+    MockitoAnnotations.initMocks(this);
+    when(mockschedular.scheduleWithFixedDelay(any(Runnable.class), anyLong(), anyLong(), any(TimeUnit.class)))
+        .thenReturn(futureTaskMock);
+  }
+
+  @After
+  public void afterTest() throws IOException
+  {
+    cleanTable();
+    FileUtils.deleteDirectory(new File(dir));
+  }
+
+  @Test
+  public void testDBPoller() throws InterruptedException
+  {
+    insertEvents(10, true, 0);
+
+    JdbcStore store = new JdbcStore();
+    store.setDatabaseDriver(DB_DRIVER);
+    store.setDatabaseUrl(URL);
+
+    List<FieldInfo> fieldInfos = getFieldInfos();
+
+    Attribute.AttributeMap.DefaultAttributeMap portAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    portAttributes.put(Context.PortContext.TUPLE_CLASS, TestPOJOEvent.class);
+    TestPortContext tpc = new TestPortContext(portAttributes);
+
+    JdbcPOJOPollInputOperator inputOperator = new JdbcPOJOPollInputOperator();
+    inputOperator.setStore(store);
+    inputOperator.setTableName(TABLE_POJO_NAME);
+    inputOperator.setKey("id");
+    inputOperator.setFieldInfos(fieldInfos);
+    inputOperator.setFetchSize(100);
+    inputOperator.setBatchSize(100);
+    inputOperator.setPartitionCount(2);
+
+    TestUtils.MockBatchedOperatorStats readerStats = new TestUtils.MockBatchedOperatorStats(2);
+    DefaultPartition<AbstractJdbcPollInputOperator<Object>> apartition = new DefaultPartition<AbstractJdbcPollInputOperator<Object>>(
+        inputOperator);
+    TestUtils.MockPartition<AbstractJdbcPollInputOperator<Object>> pseudoParttion = new TestUtils.MockPartition<>(
+        apartition, readerStats);
+    List<Partitioner.Partition<AbstractJdbcPollInputOperator<Object>>> newMocks = Lists.newArrayList();
+    newMocks.add(pseudoParttion);
+
+    Collection<com.datatorrent.api.Partitioner.Partition<AbstractJdbcPollInputOperator<Object>>> newPartitions = inputOperator
+        .definePartitions(newMocks, null);
+
```

Comment:
Please cleanup this part of the code. Keep only the necessary variables for identifying the partitions.